### ISSUE TITLE
CapyBara clicks button "More Info" and fails regardless.

### ISSUE THREAD

**Issue by yehudamakarov** [learn-co-students/return-string-data-lab-v-000#499]
pull my solution, uncomment the tests and try to pass them. good luck.

**Comment by unenlightened** [comment]
Same here. Lab is working in browser but fails at  ./spec/features/product_feature_spec.rb:14

**Comment by unenlightened** [comment]
Ok, so I changed the variable declarations from "let" to "var" with the exact same code. Then, it passed. Maybe capybara doesn't like newer ES6 features.

**Comment by katrpilar** [comment]
Tried to debug this issue for a long time and the only solution is to comment out the tests as mentioned above.

**Comment by justinddaniel** [comment]
This continues to be an issue if my recent screenshare with a student is any indication. User unenlightened above appears to have hit on the solution, though the reasons why are mysterious. Perhaps something to do with the way click_button in the test interprets ES6 syntax?

**Comment by yehudamakarov** [comment]
Do you actually want to figure this out?


Yehuda Makarov

**Comment by justinddaniel** [comment]
Hi Yehuda! I would love to figure it out, it is just a matter of finding a decent block of time. I suspect that poring through the docs on how the click_button test method works will help us find the answer.

**Comment by yehudamakarov** [comment]
If we know that there is for sure a difference with `var` vs. `const` and `let` then that may be a good place to start.

**Comment by atxrenegade** [comment]
I've had this same issue in previous labs. It's incredibly misleading to teach student NEVER TO USE VAR, and then require it to pass tests written in legacy code.

**Comment by Naomi-Dennis** [comment]
The bug is still there. 

The test will pass if all JavaScript code is PRE es6. 

For example, if you're using `let` instead of `var`, or the defining an anonymous function like this: 
`(arg) => { return arg }`  instead of like this: `function(arg){ return arg}` the test will fail. 

I think this is happening because Rails doesn't have an official way of reading ES6 JavaScript. The common way is to process JavaScript with the sprockets gem, but even that is still in beta and doesn't have an official release (as of April 2019, it doesn't really work with Rails). You can read more about it here:  [Sprockets & ES6](https://github.com/rails/sprockets/blob/master/UPGRADING.md)

TL;DR: DON'T USE ES6 JavaScript FOR THIS LAB

**Comment by Samoth112** [comment]
I'm also having an issue with this lab. It looks like the test is not responding to the "click_button" method, as the error message I'm getting seems to indicate that "expect(page).to have_content product.description" is running while the truncated message is still on the page: Failure/Error: expect(page).to have_content product.description
       expected to find text "This is a test description with more text than should be there." in "Flatiron Widgets Store\nTest Product\nThis is a test description ...\nMore Info"
